### PR TITLE
Change rackId field to a string

### PR DESCRIPTION
**Pull request by cgtz**
Change the rack ID field from a long to a string to flexibly support failure-zone identifiers.

### CODE REVIEW

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java [hunk 1]
```diff
@@ -129,8 +128,8 @@ public static String getInstanceName(String host, Integer port) {
    * @param instanceConfig the {@link InstanceConfig} associated with the interested instance.
    * @return the rack id associated with the given instance.
    */
-  static Long getRackId(InstanceConfig instanceConfig) {
-    return Long.valueOf(instanceConfig.getRecord().getSimpleField(RACKID_STR));
+  static String getRackId(InstanceConfig instanceConfig) {
+    return instanceConfig.getRecord().getSimpleField(RACKID_STR);
```
Comment (pnarayanan) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java]: not sure why this does not show up as a conflict now that #896 is merged, but please take a look.

Comment (cgtz) [reply ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java]: I think it is conflicting now. I will fix the conflicts, though

**Comment by cgtz** [comment]
One note, this will not break with current clustermaps because the JSONObject.optString method supports converting non string values (like longs) to strings

**Comment by dharju** [comment]
lgtm

**Comment by cgtz** [comment]
@pnarayanan nothing else to add to this patch
feel free to merge

**Pull request merged**